src/test/java/io/confluent/idesidecar/restapi/resources/ConnectionsResourceTest.java

```diff
@@ -469,23 +477,112 @@ void listConnections_returnsAllConnections() {
     assertConnectionList(expectedList, actualList);
   }
 
-  @ParameterizedTest
-  @NullAndEmptySource
-  void updateConnectionWithNullOrEmptyIdShouldFail(String id) {
-    var originalId = "c1";
-    var connectionSpec = ccloudTestUtil.createConnection(
-        originalId, "Connection 1", ConnectionType.LOCAL);
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("invalidConnectionUpdateParams")
+  void updateConnectionViaPatchWithInvalidInputShouldFail(
+      String testName,
+      String requestId,
+      String requestName,
+      String requestType,
+      String expectedErrorSource,
+      String expectedErrorDetail) {
+
+    var connectionId = "c1";
+    var connectionName = "Connection 1";
+    var connectionType = ConnectionType.CCLOUD;
+
+    // Create authenticated connection
+    ccloudTestUtil.createAuthedConnection(connectionId, connectionName, connectionType);
+
+    // Log current state if testing type change
+    if (testName.contains("Change Type")) {
+      var currentConnection = given()
+          .when()
+          .get("/gateway/v1/connections/{id}", connectionId)
+          .then()
+          .extract()
+          .response();
+      System.out.println("Current connection before PATCH: " + currentConnection.asString());
+    }
+
+    var patchRequest = String.format("""
+            {"id": %s, "name": %s, "type": %s}
+            """,
+        requestId == null ? "null" : "\"" + requestId + "\"",
+        requestName == null ? "null" : "\"" + requestName + "\"",
+        requestType == null ? "null" : "\"" + requestType + "\"");
```

Comment:
Glad we're using `JsonMergePatch` not `JsonPatch` :)

Reply:
What about submitting "partial updates", which don't hold the keys that should not be changed, to the `PATCH` endpoint? For instance, if we want to test updating the connection type, we should use a payload that looks like `{ "type": "NEW_TYPE" }` not `{ "id": "CURRENT_ID", "name": "CURRENT_NAME", "type": "NEW_TYPE" }`. Otherwise, we'd test the same request payloads as used for the `PUT` endpoint. WDYT?

You can easily create such payloads with the `ConnectionSpec` record so you don't have to build them from scratch yourself:

```suggestion
    var payload = new ConnectionSpec(
        requestId,
        requestName,
        requestType != null ? ConnectionType.valueOf(requestType) : null
    );
    var mapper = new ObjectMapper();
    var patchRequest = mapper.writeValueAsString(payload);
```

Reply:
thank you! I've updated the tests